flink-libraries/flink-table/src/main/java/org/apache/calcite/avatica/util/DateTimeUtils.java

```diff
@@ -751,15 +770,15 @@ private static int julianExtract(TimeUnitRange range, int julian) {
 				return day;
 			case DOW:
 				return (int) floorMod(julian + 1, 7) + 1; // sun=1, sat=7
+			case ISODOW:
+				return (int) floorMod(julian, 7) + 1; // mon=1, sun=7
 			case WEEK:
-				long fmofw = firstMondayOfFirstWeek(year);
-				if (julian < fmofw) {
-					fmofw = firstMondayOfFirstWeek(year - 1);
-				}
-				return (int) (julian - fmofw) / 7 + 1;
+				return getIso8601WeekNumber(julian, year, month, day);
```

Comment:
Why did you change the old `WEEK` code to `getIso8601WeekNumber`? If they return the same thing, why didn't you implement `getIso8601WeekNumber` as `(julian - fmofw) / 7 + 1;`?